src/Sarif/Visitors/PartitioningVisitor.cs

```diff
@@ -395,6 +405,53 @@ private SarifLog CreatePartitionLog(T partitionValue)
             return partitionLog;
         }
 
+        private IList<ReportingDescriptor> RemapResultsToRules(IList<ReportingDescriptor> rules, List<Result> results)
+        {
+            if (rules == null || !rules.Any() || results == null || !results.Any())
+            {
+                return null;
+            }
+
+            var referencedRulesByIndex = new Dictionary<ReportingDescriptor, int>();
+            int index = 0;
+
+            foreach (Result result in results)
```

Comment:
>result [](http://example.com/codeflow?start=28&length=6)

check if u have access to the run here, if yes, there is a GetRule method in the result